### PR TITLE
Tree coeff

### PR DESCRIPTION
**Pull request by gitpeterwind**
Additional functionalities to be used by the "fast_rotate" of MRChem. Should not change behaviour of other parts.
(also added "noalias" to the Eigen matrix multiplication: it is faster (does not buffer output))

### CODE REVIEW

File: src/treebuilders/ConvolutionCalculator.cpp [hunk 1]
```diff
@@ -339,16 +339,16 @@ template <int D> void ConvolutionCalculator<D>::tensorApplyOperComp(OperatorStat
         if (oData[i] != nullptr) {
             Eigen::Map<MatrixXd> op(oData[i], os.kp1, os.kp1);
             if (i == D - 1) { // Last dir: Add up into g
-                g += f.transpose() * op;
+                g.noalias() += f.transpose() * op;
```
Comment (stigrj) [inline src/treebuilders/ConvolutionCalculator.cpp:342]: Nice catch. Does this make a difference?

Comment (gitpeterwind) [reply src/treebuilders/ConvolutionCalculator.cpp:342]: I did not test so much, but I think specially the Coulomb potential is significantly faster.

Comment (stigrj) [reply src/treebuilders/ConvolutionCalculator.cpp:342]: Excellent! Have you investigated the effect of the `transpose()`?

**Review by stigrj — APPROVED** [review]
Good stuff

File: src/treebuilders/ConvolutionCalculator.cpp (hunk 1 above)
Comment (gitpeterwind) [reply src/treebuilders/ConvolutionCalculator.cpp:342]: The transpose is fine: it is a "soft" transpose, i.e. it does not actually move any data.

**Pull request merged**